### PR TITLE
Added Prime Factor [Java]

### LINKED ISSUE #283: Prime Factor [Java]
#### This issue is about [Prime Factor ](https://en.wikipedia.org/wiki/Prime_factor).

- [x] Issue Title is in form `Algo/DS Name [Language]`.
- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This Algo/DS is worth adding to this repository.
- [x] Assign this issue to me (I want to work on this).
- [x] I will send my PR only after I'm assigned to this issue by any maintainer.

### PR DESCRIPTION
**Pull request by rashinanwani**
Fixes #283

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

File: prime_factor/PrimeFactor.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import java.util.*;
+
+public class PrimeFactor {
+    public static ArrayList<Integer> primeFactor(int n)
+    {
```
Comment (Monal5031) [inline prime_factor/PrimeFactor.java]: Opening bracket should be on same line. Please add this `{` after `)` on line 4.

File: prime_factor/PrimeFactor.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        ArrayList<Integer> primeNo = new ArrayList<Integer>();
```
Comment (Monal5031) [inline prime_factor/PrimeFactor.java]: This should be `ArrayList<Integer> primeNo = new ArrayList();`

File: prime_factor/PrimeFactor.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        for (int i = 2; i <=n; i++)
```
Comment (Monal5031) [inline prime_factor/PrimeFactor.java]: add space between `<=` and `n`. Also correct the indentations here

File: prime_factor/PrimeFactor.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        {
+            if ( n % i == 0)
```
Comment (Monal5031) [inline prime_factor/PrimeFactor.java]: correct indentations here.

File: prime_factor/PrimeFactor.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+                    n = n / i;
+                }
+            }
+        }
+        return primeNo;
+    }
+
+    public static void main(String[] args) {
+        int n = 8;
+        ArrayList<Integer> primes = primeFactor(n);
```
Comment (Monal5031) [inline prime_factor/PrimeFactor.java]: This line is not required, please remove it.

File: prime_factor/PrimeFactor.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+        System.out.println("Prime Factors are:");
+        for (Integer i : primeFactor(n))
```
Comment (Monal5031) [inline prime_factor/PrimeFactor.java]: Correct indentations here.

File: prime_factor/PrimeFactor.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+        {
+            System.out.println(i);
+        }
+    }}
```
Comment (Monal5031) [inline prime_factor/PrimeFactor.java]: The last `}` should go on next line. Also add a blank line at end.

File: README.md [hunk 8]
```diff
@@ -32,7 +32,7 @@ Community (college) maintained list of Algorithms and Data Structures implementa
...
 | [Modular Exponential](http://www.geeksforgeeks.org/modular-exponentiation-power-in-modular-arithmetic/) | [:white_check_mark:](modular_exponential/modular_exponential.c) | | [:white_check_mark:](modular_exponential/ModularExponential.java) | [:white_check_mark:](modular_exponential/modular_exponential.py) | [:white_check_mark:](modular_exponential/modular_exponential.go) | |
 | [N-Queen Problem](https://en.wikipedia.org/wiki/Eight_queens_puzzle) | | [:white_check_mark:](n_queen_problem/NQueenProblem.cpp) | [:white_check_mark:](n_queen_problem/NQueenProblem.java) | [:white_check_mark:](n_queen_problem/n_queen_problem.py) | | |
-| [Prime Factor](https://en.wikipedia.org/wiki/Prime_factor) | [:white_check_mark:](prime_factor/prime_factor.c) | | | | | |
+| [Prime Factor](https://en.wikipedia.org/wiki/Prime_factor) | [:white_check_mark:](prime_factor/prime_factor.c) | |[:white_check_mark:](prime_factor/PrimeFactor.java) | | | |
```
Comment (Monal5031) [inline README.md]: Please add space between `|` and `[`

File: prime_factor/PrimeFactor.java [hunk 9]
```diff
@@ -0,0 +1,28 @@
+import java.util.*;
```
Comment (Monal5031) [inline prime_factor/PrimeFactor.java]: import only what you need.

File: prime_factor/PrimeFactor.java [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+        int n = 8;
+        System.out.println("Prime Factors are:");
+        for (Integer i : primeFactor(n)) {
+            System.out.println(i);
+        }
+    }
+}
+
+
+
```
Comment (Monal5031) [inline prime_factor/PrimeFactor.java]: Remove these extra blank lines at end.

File: prime_factor/PrimeFactor.java [hunk 11]
```diff
@@ -0,0 +1,28 @@
+import java.util.Arrays;
```
Comment (Monal5031) [inline prime_factor/PrimeFactor.java]: Remove this line

File: prime_factor/PrimeFactor.java [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+            if ( n % i == 0) {
+                primeNo.add(i);
+                while (n % i == 0) {
+                    n = n / i;
+                }
+            }
+        }
+        return primeNo;
+    }
+public static void main(String[] args) {
```
Comment (aviaryan) [inline prime_factor/PrimeFactor.java]: Please indent this line.

File: prime_factor/PrimeFactor.java [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+    }
+
+    public static void main(String[] args) {
+        int n = 8;
+        System.out.println("Prime Factors are:");
+        for (Integer i : primeFactor(n)) {
+            System.out.println(i);
+        }
+    }
+}
```
Comment (Monal5031) [inline prime_factor/PrimeFactor.java]: Add a newline at end, also remove the extra spaces after this last `}`.

**Comment by aviaryan** [comment]
@rashinanwani Thanks for this PR. There was a [minor issue](https://github.com/iiitv/algos/pull/305/commits/0d7a3b0699a19657ce2aedd6f86a77c3ca5c86b2) which I took the liberty of correcting. Expecting more PRs from you. 
Cheers.

**Pull request merged**